internal/gengapic/gengapic.go

```diff
@@ -88,17 +88,20 @@ func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, er
 				return &g.resp, errors.E(nil, "error decoding service config: %v", err)
 			}
 		case "grpc-service-config":
-			data, err := os.Open(s[e+1:])
+			f, err := os.Open(s[e+1:])
 			if err != nil {
 				return &g.resp, errors.E(nil, "error opening gRPC service config: %v", err)
 			}
 
-			cpb := conf.ServiceConfig{}
-			err = jsonpb.Unmarshal(data, &cpb)
+			data, err := ioutil.ReadAll(f)
+			if err != nil {
+				return &g.resp, errors.E(nil, "error reading gRPC service config: %v", err)
+			}
+
+			g.grpcConf, err = conf.New(data)
```

Comment:
Minor: seems like we should pass the file directly (as an `io.Reader`) rather than having to `ioutil.ReadAll` first.

Reply:
Done.